hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/ComplexAvroKeyGenerator.java

```diff
@@ -32,7 +34,8 @@ public class ComplexAvroKeyGenerator extends BaseKeyGenerator {
 
   public ComplexAvroKeyGenerator(TypedProperties props) {
     super(props);
-    this.recordKeyFields = Arrays.stream(props.getString(KeyGeneratorOptions.RECORDKEY_FIELD_NAME.key()).split(","))
+    this.recordKeyFields = doAutoGenerateRecordKeys() ? Collections.emptyList() :
```

Comment:
we have to be a bit careful about setting this to empty list. This list is passed as-is to the `KeyGenUtils#getRecordKey` taht could fail for empty list.

Reply:
when auto generation of record key is enabled, we should not be invoking KeyGenUtils#getRecordKey since keys are auto generated. so, I guess we should be ok.